maestro-app/src/main/java/bio/overture/maestro/app/infra/adapter/inbound/webapi/ManagementController.java

```diff
@@ -35,21 +32,45 @@ public ManagementController(Indexer indexer) {
         return Mono.just(response);
     }
 
+    @DeleteMapping("/index/repository/{repositoryCode}/study/{studyId}/analysis/{analysisId}")
+    @ResponseStatus(HttpStatus.OK)
+    public Mono<IndexResult> removeAnalysis(@PathVariable String analysisId,
+                                            @PathVariable String studyId,
+                                            @PathVariable String repositoryCode) {
+        log.debug("in removeAnalysis, args studyId {}, repoId: {}, analysisId : {}", studyId,
+            repositoryCode, analysisId);
+        return indexer.removeAnalysis(RemoveAnalysisCommand.builder()
+            .analysisIdentifier(
+                AnalysisIdentifier.builder()
+                    .repositoryCode(repositoryCode)
+                    .analysisId(analysisId)
+                    .studyId(studyId)
+                    .build()
```

Comment:
You don't have to make the change but just something to consider, if you notice a pattern where you only ever call a method having just called `.build()` on a builder in many different places, maybe have it just accept the builder instead?